apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextSnapshot.java

```diff
@@ -49,12 +49,15 @@
 
     private int entryApplicationInstanceId = DictionaryUtil.nullValue();
 
-    ContextSnapshot(ID traceSegmentId, int spanId, List<DistributedTraceId> distributedTraceIds) {
+    private CorrelationContext correlationContext;
+
+    ContextSnapshot(ID traceSegmentId, int spanId, List<DistributedTraceId> distributedTraceIds, CorrelationContext correlationContext) {
         this.traceSegmentId = traceSegmentId;
         this.spanId = spanId;
         if (distributedTraceIds != null) {
             this.primaryDistributedTraceId = distributedTraceIds.get(0);
         }
+        this.correlationContext = correlationContext;
```

Comment:
As you are propagating the context, and this is called a snapshot, you should clone the existing context to new thread. Otherwise, this is not a `ContextSnapshot`. Think in this way, when you do the thread snapshot, what does it mean?

Reply:
In this case, I neglected the interference between threads. Change that using `clone` mode.